src/ilasm/assembler.cpp

```diff
@@ -126,8 +126,16 @@ mdToken Assembler::ResolveClassRef(mdToken tkResScope, __in __nullterminated con
             }
             return tkRet;
         }
-        else  // needs to be resolved
-            if(!m_fIsMscorlib) tkResScope = GetBaseAsmRef();
+        else
+        { // needs to be resolved
+            if (TypeFromToken(tkResScope) == mdtAssemblyRef && RidFromToken(tkResScope) > 0)
+            {
+                // considering the aliased case '[' dottedName ']' slashedName
+                AsmManAssembly* assembly = m_pManifest->m_AsmRefLst.PEEK(RidFromToken(tkResScope) - 1);
```

Comment:
I'm sorry I'm not terribly familiar with ILAsm, so please forgive me if these are naive questions:
1) Is the RID guaranteed to be in m_AsmRefLst, i.e. do you need to check the return value of PEEK before passing it to GetAsmRef?
2) Is it possible to have multiple indirections through aliasing such that you might need to iterate until you find the non-alias?